### PR TITLE
Fix: Prevent delegate voters API from returning all accounts

### LINKED ISSUE #254: [Bug] Delegate voters API returns all accounts
### Summary

`GET /api/delegates/voters` can return every account instead of only the accounts that vote for the requested delegate.

### Steps to reproduce

1. Run an ADAMANT node that includes the delegate rank/rate fix from PR #253.
2. Request voters for a delegate public key whose vote relation list is empty or whose resolved account address list is empty:

```text
GET /api/delegates/voters?publicKey=<delegatePublicKey>
```

3. Inspect the `accounts` array in the response.

### Expected behavior

The endpoint should return only accounts that vote for the requested delegate.

If the delegate has no voters, the response should be successful with an empty `accounts` array.

Delegate-related API filters should reject invalid values consistently instead of silently returning misleading data.

### Actual behavior

The voters endpoint can pass an address `$in` filter into account lookup code that does not understand that filter shape. The account lookup then drops the address filter and queries all accounts, so the API can report every account as a voter.

Related edge cases found during the audit:

- `GET /api/delegates/getNextForgers?limit=abc` silently behaves like an empty result instead of rejecting an invalid limit.
- `GET /api/accounts/delegates` does not return errors from delegate listing and can continue with an invalid intermediate result.

### Screenshots or videos

Not applicable.

### Environment

- Repository: `Adamant-im/adamant`
- Branch: `dev`
- Affected area: public delegate/account API endpoints
- Confirmed live symptom on a node running commit `6d1863aad20ef7bef377bb8103018ef7afaec9dc`, where `/api/delegates/voters` returned more than 100,000 accounts for one delegate.

### Related PR

- https://github.com/Adamant-im/adamant/pull/255

### PR DESCRIPTION
**Pull request by dev-adamant-im**
## Description

This PR fixes a delegate API regression where `GET /api/delegates/voters` could return all accounts instead of only voters for the requested delegate.

The root cause is that delegate voters are resolved through `modules.accounts.getAccounts({ address: { $in: addresses } })`, while `logic/account#getAll` only handled a string `address` filter. The unsupported `$in` filter was removed before the SQL query, so an empty effective filter could scan all accounts.

This PR also tightens related delegate API edge cases found during the endpoint audit:

- support address `$in` filters in `Account#getAll`, including uppercase normalization and empty-list handling;
- validate `GET /api/delegates/getNextForgers` `limit` as an integer between `1` and `constants.activeDelegates`;
- propagate `GET /api/accounts/delegates` delegate-listing errors instead of continuing with an invalid result;
- add regression and edge-case coverage for delegate voters, delegate lookup, next forgers, account delegate lookup, and account address list filtering.

## Related issue

Closes #254

## How to test

Executed:

```text
PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm run test:single -- test/unit/logic/account.js
PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm run test:single -- test/unit/modules/delegates.js
PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm run test:single -- test/unit/modules/accounts.js
PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm run eslint
git diff --check
```

Results:

- `test/unit/logic/account.js`: 33 passing
- `test/unit/modules/delegates.js`: 33 passing
- `test/unit/modules/accounts.js`: 45 passing
- `npm run eslint`: passed
- `git diff --check`: passed

Not completed:

- `test/api/delegates.js` was updated with HTTP-level regression coverage, but the local API testnet run was not completed because the local test database began rebuilding/syncing from peers and the funded account setup did not become stable during the run. The node was stopped via graceful shutdown.

## Risk areas

- Public API response correctness for delegate voters and next forgers.
- Account query filtering in `logic/account#getAll`.
- No consensus, serialization, transaction validity, reward, fee, delegate ordering, or round-accounting behavior is changed.

## Checklist

- [x] Repository output is English-only.
- [x] Public API validation remains strict.
- [x] No consensus-critical behavior is changed.
- [x] Tests added for regression and edge cases.
- [x] Targeted tests and lint were run.

### CODE REVIEW

**Review by adamantmm — APPROVED** [review]
Good finding

File: test/unit/modules/delegates.js [hunk 1]
```diff
@@ -233,6 +247,33 @@ describe('delegates', function () {
...
+          expect(err).to.equal('Value 0 is less than minimum 1');
+          done();
+        });
+      });
+
+      it('should reject a limit above the active delegate count', (done) => {
+        const body = { limit: constants.activeDelegates + 1 };
+        delegates.shared.getNextForgers({ body }, (err, response) => {
+          expect(response).not.to.exist;
+          expect(err).to.equal('Value 102 is greater than maximum 101');
```
[the remark this answers is not in the document]
Comment (dev-adamant-im) [reply test/unit/modules/delegates.js]: Fixed locally. The expected validation error now derives both the attempted value and the maximum from `constants.activeDelegates`, so the test stays aligned with the configured active delegate count.

Validation passed:

```text
PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm run test:single -- test/unit/modules/delegates.js
PATH="/opt/homebrew/opt/node@22/bin:$PATH" npm run eslint
git diff --check
```
